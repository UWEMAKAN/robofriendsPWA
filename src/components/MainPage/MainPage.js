import React, { Component } from 'react';

import classes from './MainPage.module.css';
import './MainPage.module.css';
import SearchBox from '../SearchBox/SearchBox';
import CardList from '../CardList/CardList';
import Scroll from '../Scroll/Scroll';
import CounterButton from '../CounterButton/CounterButton';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';


class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobot();
  }
  
  filterRobots = () => {
    return this.props.robots.filter(robot => robot.name.toLowerCase().includes(this.props.searchField.toLowerCase()));
  }

  render() {
    const { isPending, onSearchChange } = this.props;
    return isPending ?
      <div className={ `${ classes.Loading } tc` }>
        <h1 className={ `${ classes.Heading } f1` } >LOADING...</h1>
      </div> :
      <div className='tc'>
        <h1 className={ `${ classes.Heading } f1` }>RoboFriends</h1>
        <SearchBox changed={ onSearchChange }/>
        <CounterButton color='steelblue' />
        <Scroll>
          <ErrorBoundary>
            <CardList robots={ this.filterRobots() }/>
          </ErrorBoundary>
        </Scroll>
      </div>
  }
}

export default MainPage;
