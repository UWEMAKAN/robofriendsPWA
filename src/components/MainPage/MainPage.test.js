import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';
import classes from './MainPage.module.css';

import {createSerializer} from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

let wrapper;

beforeEach(() => {
  const mockProps = {
    robots: [],
    isPending: false,
    searchField: '',
    onSearchChange: jest.fn(),
    onRequestRobot: jest.fn()
  }
  wrapper = shallow(<MainPage { ...mockProps }/>);
});

describe('MainPage Container Component', () => {
  it('expect to render MainPage Container Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('expect to render MainPage correcting for isPending: true', () => {
    const mockProps3 = {
      robots: [ { id: 2, name: 'john', email: 'john@gmail.com' } ],
      isPending: true,
      searchField: 'john',
      onSearchChange: jest.fn(),
      onRequestRobot: jest.fn()
    };
    const wrapper3 = shallow(<MainPage { ...mockProps3 } />);
    expect(wrapper3.instance().render()).toEqual(
      <div className={ `${ classes.Loading } tc` }>
        <h1 className={ `${ classes.Heading } f1` } >LOADING...</h1>
      </div>
    );
  });
  describe('expect to filter robots correctly', () => {
    it('when searchField matches', () => {
      expect(wrapper.instance().filterRobots()).toEqual([]);
      const mockProps2 = {
        robots: [ { id: 2, name: 'john', email: 'john@gmail.com' } ],
        isPending: false,
        searchField: 'john',
        onSearchChange: jest.fn(),
        onRequestRobot: jest.fn()
      };
      const wrapper2 = shallow(<MainPage { ...mockProps2 } />);
      expect(wrapper2.instance().filterRobots()).toEqual([ { id: 2, name: 'john', email: 'john@gmail.com' } ]);
    });
    it('when searchField does not match', () => {
      expect(wrapper.instance().filterRobots()).toEqual([]);
      const mockProps3 = {
        robots: [ { id: 3, name: 'john', email: 'john@gmail.com' } ],
        isPending: false,
        searchField: 'a',
        onSearchChange: jest.fn(),
        onRequestRobot: jest.fn()
      };
      const wrapper3 = shallow(<MainPage { ...mockProps3 } />);
      expect(wrapper3.instance().filterRobots()).toEqual([]);
    });  
  });  
});