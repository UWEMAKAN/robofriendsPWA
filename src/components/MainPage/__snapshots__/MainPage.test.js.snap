// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Container Component expect to render MainPage Container Component 1`] = `
<div
  className="tc"
>
  <h1
    className="Heading f1"
  >
    RoboFriends
  </h1>
  <searchBox
    changed={[MockFunction]}
  />
  <CounterButton
    color="steelblue"
  />
  <scroll>
    <ErrorBoundary>
      <cardList
        robots={Array []}
      />
    </ErrorBoundary>
  </scroll>
</div>
`;
